src/main/java/utils/DriverManager.java

```diff
@@ -3,56 +3,127 @@
 import configurations.ConfigReader;
 import io.appium.java_client.AppiumDriver;
 import io.appium.java_client.android.AndroidDriver;
+import io.appium.java_client.ios.IOSDriver;
+import io.appium.java_client.ios.options.XCUITestOptions;
 import org.openqa.selenium.remote.DesiredCapabilities;
 
 import java.net.URI;
 import java.time.Duration;
 
 public class DriverManager {
-    private static AppiumDriver driver;
+    private static final ThreadLocal<AppiumDriver> driverThreadLocal = new ThreadLocal<>();
+    private static final Duration IMPLICIT_WAIT_TIMEOUT = Duration.ofSeconds(20);
 
-    private static final String DEVICE_NAME = ConfigReader.emulatorConfig.deviceName();
-    private static final String PLATFORM_NAME = ConfigReader.emulatorConfig.platformName();
-    private static final String APP_PACKAGE = ConfigReader.emulatorConfig.appPackage();
-    private static final String APP_ACTIVITY = ConfigReader.emulatorConfig.appActivity();
-    private static final String URL = ConfigReader.emulatorConfig.remoteURL();
-    private static final String AUTOMATION_NAME = ConfigReader.emulatorConfig.automationName();
-
-    private static DriverManager INSTANCE;
-
-    public static DriverManager getInstance() {
-        if (INSTANCE == null) {
-            INSTANCE = new DriverManager();
+    public static AppiumDriver getDriver() {
+        AppiumDriver driver = driverThreadLocal.get();
+        if (driver == null) {
+            driver = initDriver();
+            driverThreadLocal.set(driver);
         }
-        return INSTANCE;
+        return driver;
     }
 
-    public AppiumDriver initDriver() throws Exception {
+    private static AppiumDriver initDriver() {
         DesiredCapabilities capabilities = new DesiredCapabilities();
+        String appiumUrl = getAppiumUrl();
 
-        capabilities.setCapability("deviceName", DEVICE_NAME);
-        capabilities.setCapability("platformName", PLATFORM_NAME);
-        capabilities.setCapability("appPackage", APP_PACKAGE);
-        capabilities.setCapability("appActivity", APP_ACTIVITY);
-        capabilities.setCapability("automationName", AUTOMATION_NAME);
-
-        driver = new AndroidDriver(new URI(URL).toURL(), capabilities);
-        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(20));
+        DriverFactory driverFactory = createDriverFactory();
+        AppiumDriver driver = driverFactory.createDriver(appiumUrl, capabilities);
 
+        driver.manage().timeouts().implicitlyWait(IMPLICIT_WAIT_TIMEOUT);
         return driver;
     }
 
-    public void quitDriver() {
-        if (driver != null) {
-            driver.quit();
+    private static DriverFactory createDriverFactory() {
+        String platformName = ConfigReader.ENVIRONMENT_CONFIG.os().toLowerCase();
+        if (platformName.contains("android")) {
```

Comment:
You will have to check if it's android or ios quite often so id recommend to create a util class for that